packages/kit/src/runtime/client/renderer.js

```diff
@@ -282,9 +282,7 @@ export class Renderer {
 		await 0;
 
 		// After `await 0`, the onMount() function in the component executed.
-		// If there was no scrolling happening (checked via pageYOffset),
-		// continue on our custom scroll handling
-		if (pageYOffset === 0 && opts) {
```

Comment:
Please could someone enlighten me on this. I'm not sure about the intent of limiting scroll function to when we're at the top of the page. Why not persit scroll in any scroll position?

Reply:
`pageYOffset === 0` is used to check if the dev did manual scrolling in an `onMount()` function, if they did we shouldn't apply our scrolling logic. It's not the best heuristic but I haven't found a better way around it yet.

Reply:
FYI there's also another fix that touches that in #2735